AEPServices/Sources/ui/MessageMonitorService.swift

```diff
@@ -0,0 +1,80 @@
+/*
+ Copyright 2020 Adobe. All rights reserved.
+ This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License. You may obtain a copy
+ of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software distributed under
+ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ OF ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ */
+
+import Foundation
+
+/// This class is used to monitor if an UI message is displayed at some point in time, currently this applies for full screen and alert messages.
+/// The status is exposed through isMessageDisplayed.
+public class MessageMonitorService {
+    private let LOG_PREFIX = "MessageMonitor"
+    private var isMsgDisplayed = false
+    internal var globalUIMessagingListener: GlobalUIMessaging?
+    private let messageQueue = DispatchQueue(label: "com.adobe.uiservice.messagemonitor")
+
+    /// GlobalUIMessaging listener which is used to listen for message visibility updates.
+    public func setGlobalUIMessagingListener(listener: GlobalUIMessaging?) {
```

Comment:
Currently, to set the messaging delegate we have to do:
`ServiceProvider.shared.messageMonitorService.setGlobalUIMessagingListener(listener: myListener)`
To follow a more iOS-like pattern, let's remove this setter and the variable `globalUIMessagingListener` from this class. 

Instead, add a public variable to `AEPCore`  called `messgaingDelegate` of type `GlobalMessagingDelegate`.

Then the workflow for setting a messaging delegate `AEPCore.messgaingDelegate = myDelegate`.